src/publicApi.ts

```diff
@@ -300,14 +300,15 @@ export interface WorkspaceData {
     phase: WorkspacePhase;
     description: string;
     lastUsed: Date;
-    recentFolders : string[];
+    recentFolders: string[];
 }
 
 export function rawWorkspaceToWorkspaceData(rawWorkspaces: Workspace): WorkspaceData;
 export function rawWorkspaceToWorkspaceData(rawWorkspaces: Workspace[]): WorkspaceData[];
 export function rawWorkspaceToWorkspaceData(rawWorkspaces: Workspace | Workspace[]) {
     const toWorkspaceData = (ws: Workspace) => {
-        const url = new URL(ws.context!.contextUrl);
+        const contextUrl = ws.context?.details.case === 'git' ? ws.context.details.value.normalizedContextUrl : ws.context.contextUrl;
```

Comment:
I don't think we should ever use context URL on UI, but either:
- ensure that Public API always return normalized context URL according to the logic in https://github.com/gitpod-io/gitpod/blob/0bb6aaad9d4923d5510de865437f22c7f726833f/components/gitpod-protocol/src/context-url.ts#L31C28-L31C28
- or call to https://github.com/gitpod-io/gitpod/blob/0bb6aaad9d4923d5510de865437f22c7f726833f/components/gitpod-protocol/src/context-url.ts#L31C28-L31C28 from here 

we should also filter out or return undefined if it cannot be recovered

Reply:
i.e ws.context.contextUrl should be used as is

Reply:
> ensure that Public API always return normalized context URL according to the logic in

Public API defined different context, like `prebuild`, `git`, `snapshot`. But only use `git` [[1](https://github.com/gitpod-io/gitpod/blob/7e0c605a3d470b8cee0e841e51da9b20022f4f4b/components/public-api-server/pkg/apiv1/workspace.go#L332-L335)] right now.

We can filter that it only return `git` type and use `normalizedContextUrl`? To call server protocol again looks weird

Reply:
`context URL` is a part of UI now, if we remove this option, then we can remove those properties

<img width="458" alt="image" src="https://github.com/gitpod-io/gitpod-vscode-desktop/assets/20944364/14f3c53b-885b-4623-aecf-3786a1507b62">

Reply:
For now to work around maybe the easiest, call https://github.com/gitpod-io/gitpod/blob/0bb6aaad9d4923d5510de865437f22c7f726833f/components/gitpod-protocol/src/context-url.ts#L31C28-L31C28 pass fake objet without normalizedContextUrl but context url. You have gitpod-protocol already in the extension.

Later we need to talk with @geropl @svenefftinge to align how we want to deal context URLs. It maybe we should rather parse in server metadata from context URL, store them and fetch here instead of trying to derive from the URL at all.